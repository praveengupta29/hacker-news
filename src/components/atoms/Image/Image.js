import React from 'react';
import PropTypes from 'prop-types';

const Image = ({ alt, className, src, srcset, sizes, ...other }) => (
  <img
    {...other}
    alt={alt}
    className={className}
    src={src}
    srcSet={srcset || null}
    sizes={sizes || null}
  />
);

Image.defaultProps = {
  className: '',
  sizes: '',
  srcset: '',
};

Image.propTypes = {
  alt: PropTypes.string.isRequired,
  className: PropTypes.string,
  src: PropTypes.string.isRequired,
  srcset: PropTypes.string,
  sizes: PropTypes.string,
};

export default Image;
