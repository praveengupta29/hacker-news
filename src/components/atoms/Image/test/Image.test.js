import React from 'react';
import { shallow } from 'enzyme';
import Image from '../Image';

describe('Image Component', () => {
  const props = {
    alt: 'test',
    src: '/',
    className: 'test',
  };

  test('should render correctly', () => {
    const icon = shallow(<Image {...props} />);
    expect(icon).toMatchSnapshot();
  });
});
