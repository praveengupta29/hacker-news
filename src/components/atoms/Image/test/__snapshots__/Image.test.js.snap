// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Component should render correctly 1`] = `
<img
  alt="test"
  className="test"
  sizes={null}
  src="/"
  srcSet={null}
/>
`;
